e2e/nomostest/nt.go

```diff
@@ -80,11 +88,8 @@ type NT struct {
 	// Config specifies how to create a new connection to the cluster.
 	Config *rest.Config
 
-	// Client is the underlying client used to talk to the Kubernetes cluster.
-	//
-	// Most tests shouldn't need to talk directly to this, unless simulating
-	// direct interactions with the API Server.
-	Client client.Client
+	// KubeClient is a test wrapper used to make Kubernetes calls.
+	KubeClient *testkubeclient.KubeClient
```

Comment:
I could have not renamed the field, but it seemed like `Client` was too generic anyway, since the tests talk to multiple services (git, oci, k8s, gcp, etc). And the wrapper uses some of the same method names as the client.Client (Get, List, etc), which Go doesn't allow when embedded.

Reply:
I had in mind removing `Client` as a field here and replacing it with a `KubeClient` embedding

Reply:
Oh, you mean embedding it in the NT? That seems confusing from a polymorphic standpoint. The NT is a "Test" not a "Client".

Reply:
From a more tactical standpoint, while Go does support embedding multiple structs, all public and private fields and methods must be unique, which makes it hard to do in practice.